src/client.py

```diff
@@ -0,0 +1,362 @@
+        :param filter: MERMAID project resource filters.
+        :type filter: str
+        :param filter_val: (optional) Required for resource filters requiring values. eg. (size_min/size_max,
+        count_min/count_max, length_min/length_max, len_surveyed_min/len_surveyed_max,
+        sample_date_before/sample_date_after).
+        :type filter_val: str, int
+        :param resource: MERMAID project entity resources.
+        :return: projects path, returns None if invalid project name or id.
+        :rtype: str, None
+        """
+        # Generate path.
+        path = None
+        if resource:
+            if id:
+                path = 'projects/' + id + '/' + resource + '/'
+            elif name:
+                p_id = self.get_project_id(name=name)
+                # Nested if instead of elif name and self.get_project_id(name=name) to improve performance by reducing
+                # the number of API calls for name or id NoneType check.
+                if p_id:
+                    path = 'projects/' + p_id + '/' + resource + '/'
+        else:
+            if id:
+                path = 'projects/' + id + '/'
+            elif name:
+                p_id = self.get_project_id(name=name)
+                if p_id:
+                    path = 'projects/' + p_id + '/'
+
+        # API call and filters check.
+        if path:
+            if filter and filter_val:
+                payload = {filter: filter_val}
+                return self.api_root(path, parameters=payload)
+            elif filter:
+                return self.api_root(path, parameters=filter)
+            else:
+                return self.api_root(path)
+        return path
+
+    # Get functions.
+    def get_info(self, info):
+        """
+        Gets non-project resource information. Authentication not required for health, version, profiles, projecttags,
+        sites, managements. Authentication required for me.
+        See https://mermaid-api.readthedocs.io/en/latest/nonproject.html for more information on non-project resources.
+        :param info: health, version, profiles, projecttags, sites, managements, me.
+        :type info: str
+        :return: non-project resource data.
+        :rtype: dict, str, None
+        """
+        # TODO: Handle pagination, filters, PUT, HEAD OPTIONS requests
+        # Authentication required for access to 'me' API endpoint.
+        if info == 'me' and not self.authenticated:
+            return None
+
+        if info in self.npr_endpoints:
+            return self.api_root(info)
+        return None
+
+    def get_choices(self):
+        """
+        Gets a list of objects, each one of which has a name item (e.g., countries) and a data item that is a list
+        of available choice objects.
+        :return: list of choice objects.
+        :rtype: list
+        """
+        # TODO: Filter: /updates/?timestamp="DATE"
+        return self.api_root('choices')
+
+    def get_attribute(self, attr):
```

Comment:
This method name should really be plural... but really I'd just split it into one method for each endpoint (i.e. `get_benthic_attributes` etc.).